File: src/main/resources/Exercise/Exo5/mainExo.php
<?php
	include 'soluceExo.php';
	include 'userExo.php';

	function generateRandomArray($minSize, $maxSize, $minValue, $maxValue){
		$size = rand($minSize, $maxSize);
		for($i = 0; $i < $size; $i++){
			$array[$i] = rand($minValue, $maxValue);
		}
		return $array;
	}

	function run_test($array, $size){
		$result1 = arraySum1($array, $size);
		$result2 = arraySum($array, $size);
		if($result1 == $result2){
			echo "Test passed";
			return 0;
		}
		else{
			echo "Test failed";
			return 1;
		}
	}

	function main(){
		$failed = 0;
		$minSize = 5;
		$maxSize = 100;
		$minValue = -1000;
		$maxValue = 1000;
		$array1 = generateRandomArray($minSize, $maxSize, $minValue, $maxValue);
		$failed |= run_test($array1,count($array1));
		$array2 = generateRandomArray($minSize, $maxSize, $minValue, $maxValue);
		$failed |= run_test($array2,count($array2));
		$array3 = generateRandomArray($minSize, $maxSize, 0, 0);
		$failed |= run_test($array3,count($array3));
		$array4 = [];
		$failed |= run_test($array4,count($array4));
		return $failed;
	}

	main();
?>

File: src/main/resources/Exercise/Exo3/mainExo.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include "soluceExo.c"
#include "userExo.c"

int run_test(int array[], int size){
	int result1 = array_sum1(array, size);
	int result2 = array_sum(array, size);
	puts("");
	if(result1 == result2){
		printf("Test passed");
		return 0;
	}
	else{
		printf("Test failed");
		return 1;
	}
}

int* generateRandomArray(int size, int minValue, int maxValue){
	int* array = malloc(size*sizeof(int));
	if(array == NULL){
		exit(2);
	}
	for(int i = 0; i < size; i++){
		array[i] = rand()%(maxValue - minValue + 1) + minValue;
	}
	return array; 
}

int main(){
	srand(time(NULL));
	int minSize = 5;
	int maxSize = 95;
	int minValue = -1000;
	int maxValue = 2001;
	int failed = 0;
	int size1 = rand()%(maxSize - minSize + 1) + minSize;
	int* array1 = generateRandomArray(size1, minValue, maxValue);
	failed |= run_test(array1,size1);
	int size2 = rand()%(maxSize - minSize + 1) + minSize;
	int* array2 = generateRandomArray(size2, minValue, maxValue);
	failed |= run_test(array2,size2);
	int size3 = rand()%(maxSize - minSize + 1) + minSize;
	int* array3 = generateRandomArray(size3, 0, 0);
	failed |= run_test(array3,size3);
	int size4 = 0;
	int* array4 = generateRandomArray(size4, minValue, maxValue);
	failed |= run_test(array4,size4);
	return failed;
}